### ISSUE TITLE
Glossary Term Display in Text

### ISSUE THREAD

**Issue by nrshapiro** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#287]
Client: 2021.02.11-6202
Server: 8.6.2/2021.0201.1.Alpha
Session: d4e5c5cf-7d89-4aa1-83d6-654b0aac77db

The glossary terms now have been showing up as [Term] when highlighting is turned off, and a highlighted version of that when turned on.  That wasn't the intent of the glossary formatting preference switch.

When off, the terms should be underlined as a link when the cursor passes over them, and otherwise, should not have any indication except any formatting in the original document.

Currently On Stage Turned on:
![image](https://user-images.githubusercontent.com/4837554/107881250-8ab02b00-6eb1-11eb-9ed8-9b763fdeaaec.png)

Currently On Stage Turned Off:
![image](https://user-images.githubusercontent.com/4837554/107881267-9c91ce00-6eb1-11eb-8537-38d9733469d1.png)

As on PEP-Web GVPi, how it should behave if Glossary Formatting is turned off:
https://user-images.githubusercontent.com/4837554/107881443-6143cf00-6eb2-11eb-9c73-d2873e946eac.mp4

**Comment by bakerac4** [comment]
@nrshapiro this is due to WCAG guidelines that say color should not be the only way to mark something.

**Comment by nrshapiro** [comment]
@bakerac4 It's fine when markup is turned on, but when markup is off via the option there should be no markup other than underlining links (which is not color) when the cursor passes over as on our current site.   

Surely WCAG doesn't require us to continuously display links when the user has intentionally turned them off.  They interfere with text readability.

**Comment by nrshapiro** [comment]
@bakerac4 @adistasio 

Also, I presume the text "Glossary links are marked with []" in the upper right of the article is meant to be help text.  I'm mixed on the message there, but in any case, it should not appear if help text is turned off in the help menu.  But it still appears.  

I would also suggest making help text more distinguishable as such, giving it a unique look from other aspects of the interface so it can be identified as such.

**Comment by nrshapiro** [comment]
The help text message is removed when you turn off help text messages.
But it is still there when you turn off Glossary formatting, which doesn't make any sense...because there's no glossary formatting, so the help text is wrong.

It should only be there if formatting is turned on AND help text is turned on.

**Comment by bakerac4** [comment]
@nrshapiro I cant reproduce this on stage or dev. I also didn't change anything related to help text messages, only glossary formatting. Can you take another look after clearing your cache? (shouldn't matter though)

**Comment by nrshapiro** [comment]
@bakerac4 I just logged in on one of my ids as a means to refresh, and the message no longer appeared, whether I had glossary terms showing or not.  And I did have help turned on:

Glossary terms showing.  
<img width="1475" alt="2021-02-19_11h30_57" src="https://user-images.githubusercontent.com/4837554/108532824-43a5a980-72a6-11eb-88c7-723a6c3bccba.png">

Help Preferences
![image](https://user-images.githubusercontent.com/4837554/108532985-76e83880-72a6-11eb-8ef8-02291944b65d.png)

I'm not a big fan of the message showing there anyway, at least as formatted.  But I am not really up to speed on the requirements of WCAG.  But in any case, the message seems to be completely gone now.

**Comment by bakerac4** [comment]
@nrshapiro that text doesn't have anything to do with help (its there to satisfy wcag) so it wont be changed at all by the help text show/hide. But its also there in your picture, you mentioned you didn't like how it looked so I moved it to below the banner

**Comment by nrshapiro** [comment]
@bakerac4  - LOL, just before you wrote, I noticed that the message is there, but now it's on the left side. 

Can you direct me to the section of WCAG that requires it (a link would be helpful)?  Why would this particular link require a obtrusive message like that whereas the other link types don't? It doesn't fully make sense to me that this particular feature requires such a message, and I'd like to understand.

Also, I think that has helped me understand my objection to it...It should be in one of the "control" areas, NOT in the content area; it looks too much like part of the content/article.  It takes away from the article, which otherwise looks like the print edition.

On a smaller screen the top is pretty busy.  What about the bottom status bar area?

**Comment by bakerac4** [comment]
@nrshapiro Haha, that makes sense. Regarding the WCAG section - https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html under `Situation A: If the color of particular words, backgrounds, or other content is used to indicate information:`

I also think I read at one point that the description should be near the instances of use (which makes sense) so thats why I put it in the document itself

**Comment by nrshapiro** [comment]
@bakerac4  Thanks, that's helpful.  If I read that right, the explanation is due to the use of color.  

There's a very simple solution.  Do the links like our old system.  The way Gavant chose to do the links (we didn't ask for them to be shaded in color), is what is causing the complications.  Not only this, but the need to be able to turn off the option.

To explain further, think about my previous comment.  We don't want the content area to look like a user interface.  We want it to look like an article--a printed article (this is an ARCHIVE of printed articles).  Coloring the glossary links, takes it further from that.  And that's in fact why we needed a way to turn them off in the preferences.  We didn't need that before.  

And then, using the color, make us go a step further--you then put [] around the glossary words.  Again, further away from our printed model.

And that in fact, made ME go a step further.  I reduced the number of words that had glossary links, because it was too obtrusive. 

All of this, because of the way it was linked.

In our old system, the links show up ONLY as underlines when you hover over them.  No one has complained about that, so we had no reason to change from that model.  That's not to say that WCAG doesn't have something to say about hover links--I don't know.  But looking at Taylor and Francis's site, they have lots of hover links.

So here's what I propose.

1) Go back to the old way.  Mostly.  
    a) When Show Glossary Links is off, use the underlining to show the hover links when the cursor passes over the words.   
    b) When Show Glossary Links is on, show the underlines under the terms all the time.  
    c) Show highlight terms should be off by default.

And thus: no more brackets, no more colors, no more message.

One option for an "aesthetic" adjustment.  If you prefer, and if I read the WCAG section right, you can try to use GRAY background shading to highlight the terms when show is "on" if that eliminates the need for the brackets and message.  That's not a color that some people cannot see.

Thanks.

**Comment by bakerac4** [comment]
@nrshapiro Certainly can be done, however I would recommend against it. Just an outsider opinion, coming and using your site I found it near impossible to figure out what are glossary terms and what are not. The color helps (for 90% of people) to make it immediately visible and know that these are something I can click on.

Also re-reading some documentation, I believe links are suppose to have a color (they generally do for most websites) https://webaim.org/blog/wcag-2-0-and-link-colors/

```
So if you combine these two requirements, in order to be Level AA conformant, your page must have all of the following:

A 4.5:1 contrast between the non-link text color and the background.
A 4.5:1 contrast between the link text color and the background.
A 3:1 contrast between the link text color and the surrounding non-link text color.
In other words, your link color has to be significantly different from the background color AND the surrounding text color, which also has to be significantly different from the background color.
```

 It does have downsides, however but I don't think adding the line below the graphic is a huge deal. But if you feel strongly about this, I will remove them as requested.

**Comment by nrshapiro** [comment]
@bakerac4 

I think there are a few factors for PEP-Web you should keep in mind.

1) this is not a web page user interface, it's an article.

2) Glossary links are actually a convenience function...the user is interested in what a term means, and they want to look it up.  Instead of selecting it, and right clicking and selecting "Search the Web", or "Look up in Dictionary", we provide a more convenient means.  **They are not a required means of navigation in our website.**

Therefore, whatever we do to show glossary links all the time are a distraction from the content.  Look at other examples of online databases with articles.  You won't see a lot of distracting elements in the text.

1) Wikipedia - Links are all blue text, with a underline when you hover.  E.g., https://en.wikipedia.org/wiki/Psychoanalysis#Topographic_theory Note that these are hover links, but they are also navigation links (the hover is a preview)

2) Sage Publishing - Blue text with underline when hovering.   https://journals.sagepub.com/doi/full/10.1177/0003065119831767

3) Taylor and Francis - Blue text, at least some links show nothing until you hover, then they are blue.  
(note: they overlap our journals, but I couldn't find a free example there, so here's one in a different topic area): https://www.tandfonline.com/doi/full/10.1080/0144929X.2018.1541255  (Hover above 2011 in the section "Extension of the umbrella".  Or on other years in citations...same thing.)

While I could find more examples, I think you get the point.  These sites displaying text from articles try to tone it down enough not to be distracting.  With the exception of #3 above--I think their site is horrifically distracting (elements around the text)

Another important point to consider:

1) There are so many glossary term links in our text, most users will just turn off "Show Glossary Links".  While that solves the problem, what's the point of doing it in a way that makes them turn it off?  (And as I mentioned, it already forced us to make decisions to remove links...some of which David was fairly unhappy about).

2) Most users aren't interested in all those Glossary links...they are interested in the article, and most of them are practicing professionals and researchers.  Most of the terms will NEVER be clicked, because they don't need the "definition".  When they do click, it would be because they are interested in how the various famous glossaries compare in their definitions (since we integrate 4 or so glossaries into 1 for that very reason).

3) I think I'm one of the few who likes having an option to see the glossary links--but that's just because it helps me check my linking algorithms, and look for words that are linked that should not be.  That's not a reason at all to show them though!

Now, regarding number 1 and 3 above, I think that since we now do have a option to turn them off, we can vary a bit from our old approach and follow the other sites.  By that I mean, it would be ok to color the text the current link color (not shade it and bracket it) and underline it when you hover if the Show Glossary links is turned on.  When it's turned off, it should just underline when you hover.  

That seems to meet Internet standards and provides consistency of links.  Again, it may not meet WCAG standards--though I don't see that yet to be true.  But still happy to discuss.

And speaking of consistency, I think that means that other links in the text, like footnotes etc., should also underline when you hover.

To summarize, and amend my proposal:

a) When Show Glossary Links is off, use the underlining to show the hover links when the cursor passes over the words.
b) When Show Glossary Links is on, use a blue (or the current link color) for the text, and show the hover links when the cursor passes over the words.
c) Show highlight terms should be off by default.

No brackets or message should be necessary.

**Comment by nrshapiro** [comment]
@bakerac4  BTW You can hold off till next week on the changes above.  I will discuss the issue with the board this weekend.

**Comment by nrshapiro** [comment]
@bakerac4 Per my discussion with executive board member DT, please go ahead and make the change I describe.  People want to read...not see the glossary terms in the text (primarily).  When they'd like to see it, they'll turn on Show Glossary Links.  As I mentioned, the "show" should be off by default.

**Comment by nrshapiro** [comment]
@bakerac4  @adistasio 

- Now the show glossary links OFF behavior works correctly.  Good.
- Still not a fan of the [boxed look] when turned on.  In general, we should try not to "interfere" with the "print-like" presentation of the article.  But if it is absolutely required by WCAG, then ok (I just don't see that, but I'm not versed in it).  

But: leaving this open because I don't like the message about the markup where it is now...it is less "in your face" than before, but it interferes with the presentation of the article, looking like part of the running head.

**Comment by bakerac4** [comment]
Fixed by #303
